File: frontend/gestionale-bertozzi/src/app/models/GestioneCommesse/commessa.ts
import moment from 'moment';
import { Cliente } from '../Anagrafiche/cliente';
import { PersonaleCliente } from '../Anagrafiche/personale-cliente';
import { TipologiaCommessa } from '../Anagrafiche/tipologia-commessa';
import { StatusCommessa } from '../Anagrafiche/status-commessa';
import { PianoSviluppo } from './piano-sviluppo';

export class Commessa {
    id?: number;
    clienteId?: number;
    luogoCommessa?: string;
    progressivoCommessa?: number;
    protocollo?: string;
    pmEdileId?: string;
    referenteClienteId?: number;
    pmAmministrativoId?: string;
    tipologiaCommessaId?: number;
    descrizione?: string;
    costoAtteso?: number;
    statusCommessaId?: number;
    dataInizioPrevista?: moment.Moment;
    dataConclusionePrevista?: moment.Moment;
    utenteCreazione?: string;
    utenteModifica?: string;
    
    // Navigation properties
    cliente?: Cliente;
    referenteCliente?: PersonaleCliente;
    tipologiaCommessa?: TipologiaCommessa;
    statusCommessa?: StatusCommessa;
    pianiSviluppo?: PianoSviluppo[];

    static map(commessa: any): Commessa {
        let c = Object.assign(new Commessa(), commessa) as Commessa;
        
        // Converti le stringhe di data in oggetti moment
        if (commessa.dataInizioPrevista) {
            c.dataInizioPrevista = moment(commessa.dataInizioPrevista);
        }
        if (commessa.dataConclusionePrevista) {
            c.dataConclusionePrevista = moment(commessa.dataConclusionePrevista);
        }
        
        // Mappa le proprietà di navigazione se presenti
        if (commessa.cliente) {
            c.cliente = Cliente.map(commessa.cliente);
        }
        if (commessa.referenteCliente) {
            c.referenteCliente = PersonaleCliente.map(commessa.referenteCliente);
        }
        if (commessa.tipologiaCommessa) {
            c.tipologiaCommessa = TipologiaCommessa.map(commessa.tipologiaCommessa);
        }
        if (commessa.statusCommessa) {
            c.statusCommessa = StatusCommessa.map(commessa.statusCommessa);
        }
        if (commessa.pianiSviluppo && Array.isArray(commessa.pianiSviluppo)) {
            c.pianiSviluppo = PianoSviluppo.mapArray(commessa.pianiSviluppo);
        }
        
        return c;
    }

    static mapArray(arr: any[]): Commessa[] {
        return (arr || []).map(c => Commessa.map(c));
    }
}

File: backend/NemesiLIB/Model/GestioneCommesse/Commessa.cs
using NemesiCOMMONS.Models;
using System;
using System.Collections.Generic;

namespace NemesiLIB.Model.GestioneCommesse
{
    public class Commessa : IAuditable
    {
        public int Id { get; set; }
        public int ClienteId { get; set; }
        public string LuogoCommessa { get; set; }
        public int ProgressivoCommessa { get; set; }
        public string? Protocollo { get; set; }
        public string PmEdileId { get; set; }
        public string ReferentiCliente { get; set; }
        public string PmAmministrativoId { get; set; }
        public int TipologiaCommessaId { get; set; }
        public string Descrizione { get; set; }
        public string CommessaCodiceInterno { get; set; }
        public decimal CostoAtteso { get; set; }
        public int StatusCommessaId { get; set; }
        public DateTime? DataInizioPrevista { get; set; }
        public DateTime? DataConclusionePrevista { get; set; }
        public DateTime DataCreazione { get; set; }
        public DateTime? DataModifica { get; set; }
        public string? UtenteCreazione { get; set; }
        public string? UtenteModifica { get; set; }

        // Navigation properties
        public virtual Anagrafiche.Cliente? Cliente { get; set; }
        public virtual Anagrafiche.TipologiaCommessa? TipologiaCommessa { get; set; }
        public virtual Anagrafiche.StatusCommessa? StatusCommessa { get; set; }
        public virtual Utente? PmAmministrativo { get; set; }
        public virtual Utente? PmEdile { get; set; }
        public virtual List<PianoSviluppo>? PianiSviluppo { get; set; }
        public virtual List<OreSpeseCommessa>? OreSpese { get; set; }
    }
}